ksql-api/src/main/java/io/confluent/ksql/api/server/AcksSubscriber.java

```diff
@@ -54,22 +55,41 @@ protected void afterSubscribe(final Subscription subscription) {
   }
 
   @Override
-  public void handleValue(final JsonObject value) {
+  public void handleValue(final InsertResult result) {
     checkContext();
-    insertsStreamResponseWriter.writeInsertResponse();
-    acksSent++;
-    if (insertsSent != null && insertsSent == acksSent) {
-      close();
-    } else if (response.writeQueueFull()) {
-      if (!drainHandlerSet) {
-        response.drainHandler(v -> {
-          drainHandlerSet = false;
-          checkMakeRequest();
-        });
-        drainHandlerSet = true;
+    if (responseEnded) {
+      return;
+    }
+    if (result.succeeded()) {
+      insertsStreamResponseWriter.writeInsertResponse(new InsertAck(result.sequenceNumber()));
+      acksSent++;
+      if (insertsSent != null && insertsSent == acksSent) {
+        close();
+      } else if (response.writeQueueFull()) {
+        if (!drainHandlerSet) {
+          response.drainHandler(v -> {
+            drainHandlerSet = false;
+            checkMakeRequest();
+          });
+          drainHandlerSet = true;
+        }
+      } else {
+        checkMakeRequest();
```

Comment:
I think I've seen this snippet of code repeated quite a few times in the new API code.  Might it be worth coding this into a type?  i.e. a type that takes the response as a constructor param, and you can just call `enqueue(this::checkMakeRequest)` and it will internally handle checking if the queue is full, setting the drain handler and flag etc.?